functions.py

```diff
@@ -183,19 +184,26 @@ def get_jenkins_jobs(server, job_search_fields):
 	for i in range(fields_length):
 		fields[i] = fields[i].strip(' ')
 
-	# fetch all jobs from server
-	all_jobs = server.get_jobs()
+	# check for fields that contain valid regex
+	for field in fields:
+		print(field)
+		try:
+			# does the search field contain valid regex?
+			re.compile(field)
 
-	# parse out all jobs that do not contain any search field and/or are not OSP10, OSP13, OSP15 or OSP16 jobs
-	relevant_jobs = []
-	supported_versions = ['10', '13', '15', '16', '16.1']
-	for job in all_jobs:
-		job_name = job['name']
-		if any(supported_version in job_name for supported_version in supported_versions):
-			for field in fields:
-				if field in job_name:
+			# fetch all jobs from server that match the given regex
+			all_jobs = server.get_job_info_regex(field)
+
+			# parse out all jobs that do not contain any search field and/or are not OSP10, OSP13, OSP15 or OSP16 jobs
+			supported_versions = ['10', '13', '15', '16', '16.1']
+			for job in all_jobs:
+				job_name = job['name']
+				if any(supported_version in job_name for supported_version in supported_versions):
 					relevant_jobs.append(job)
-					break
+					print(job_name)
+
+		except re.error as e:
+			print("Error compiling regex: {}".format(e))
```

Comment:
I'm still not understanding the logic here - if `re.complie` throws an error, no jobs will appear in the report

Reply:
This isn't the case, I just tested this with the following parameters:
`job_search_fields: '[f, ^DFG-ceph-rhos'`

And it works as I think it should, here's the output:
                                        
Error compiling regex: unterminated character set at position 0 
DFG-ceph-rhos-10_director-rhel-virthost-3cont_1_to_2comp_1_to_3ceph-ipv4-vxlan-scale-up
DFG-ceph-rhos-10_director-rhel-virthost-3cont_2comp_3ceph-ipv4-vxlan-monolithic
DFG-ceph-rhos-10_director-rhel-virthost-3cont_3osdcompute-ipv4-vxlan-osdcompute-rgw
...

The report was sent to my email with the expected DFG-ceph-rhos jobs

Reply:
Ack - I'll do some tests as well